repose-aggregator/components/filters/rate-limiting-filter/src/main/java/org/openrepose/filters/ratelimiting/RateLimitingHandler.java

```diff
@@ -109,29 +115,76 @@ private boolean requestHasExpectedHeaders(HttpServletRequest request) {
         return request.getHeader(PowerApiHeader.USER) != null;
     }
 
+    private static Optional<MediaType> parseMediaType(String type) {
+        Optional<MediaType> parsedType = Optional.empty();
+        try {
+            parsedType = Optional.of(MediaType.parseMediaType(type));
+        } catch (IllegalArgumentException iae) {
+            LOG.warn("Media type could not be parsed: {}", type, iae);
+        }
+        return parsedType;
+    }
+
+    private Optional<MediaType> getPreferredMediaType(List<String> acceptValues) {
+        Optional<MediaType> preferredMediaType = Optional.of(DEFAULT_MEDIA_TYPE);
+        if (!acceptValues.isEmpty()) {
+            /*
+            Parses media types from the "Accept" header (dropping any that cannot be parsed),
+            removes any media types that are not supported by this filter (since they will not be returned anyway),
+            then sorts the remaining media types by specificity (primarily) and quality (secondarily).
+              */
+            List<MediaType> sortedRelevantAcceptTypes = acceptValues.stream()
+                    .map(RateLimitingHandler::parseMediaType)
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .filter(mediaType -> SUPPORTED_MEDIA_TYPES.stream().anyMatch(mediaType::isCompatibleWith))
+                    .sorted(new CompoundComparator<>(MediaType.SPECIFICITY_COMPARATOR, MediaType.QUALITY_VALUE_COMPARATOR))
```

Comment:
I'm guessing this is the recommended way of comparing these since there's a `sortBySpecificityAndQuality` method in `MediaType` that does essentially the same thing, but do you by chance know why?  It looks like the `SPECIFICITY_COMPARATOR` looks at the quality, then compares the number of header parameters (so in theory `*/*;what=ever` would be more important than `application/xml`), and it looks like `QUALITY_VALUE_COMPARATOR` does everything you would need.  I'm guessing I'm missing something?